pandas/_libs/parsers.pyx

```diff
@@ -614,6 +614,9 @@ cdef class TextReader:
         self.parser.cb_io = NULL
         self.parser.cb_cleanup = NULL
 
+	# TODO: the following code expects a compression string
+	if isinstnace(self.compression, dict):
+		self.compression = self.compression['method']
```

Comment:
this part seems to partly re-invent `get_handle` but doesn't support a compression dict